src/geometric_utils.ts

```diff
@@ -126,17 +214,178 @@ export class GeometricUtils {
         }
     }
 
-    public static get_nearest_point_on_bounding_box(ref_x: number, ref_y: number, spatial_payload: ULabelSpatialPayload2D, dstmax: number = Infinity): object {
-        var ret = {
+    // Return a list of polygons that define all intersections between a list of polygons
+    public static get_polygon_intersections(polygons: ULabelSpatialPayload2D[]): ULabelSpatialPayload2D[] {
+        let ret: ULabelSpatialPayload2D[] = [];
+        for (let i: number = 0; i < polygons.length; i++) {
+            for (let j: number = i+1; j < polygons.length; j++) {
+                let poly1: ULabelSpatialPayload2D = polygons[i];
+                let poly2: ULabelSpatialPayload2D = polygons[j];
+                // Ensure both polygons are closed
+                if (GeometricUtils.is_polygon_closed(poly1) && GeometricUtils.is_polygon_closed(poly2)) {
+                    let intersection: ULabelSpatialPayload2D = GeometricUtils.get_polygon_intersection_single(poly1, poly2);
+                    if (intersection != null) {       
+                        // Don't add duplicate intersections
+                        let is_duplicate: boolean = false;
+                        for (let k: number = 0; k < ret.length; k++) {
+                           if (GeometricUtils.polygons_are_equal(ret[k], intersection)) {
+                               is_duplicate = true;
+                               break;
+                           }
+                        }
+                        if (!is_duplicate) {
+                            ret.push(intersection);
+                        }
+                    }
+                }
+            }
+        }
+        return ret;
+    }
+
+    // Return the intersection of two polygons
+    public static get_polygon_intersection_single(poly1: ULabelSpatialPayload2D, poly2: ULabelSpatialPayload2D): ULabelSpatialPayload2D {
+        // Convert to turf polygons
+        try {
+            let poly1_turf = turf.polygon([poly1]);
+            let poly2_turf = turf.polygon([poly2]);
+            
+            // Find intersection
+            let intersection = turf.intersect(poly1_turf, poly2_turf);
```

Comment:
Ah, cool, so you are using Turf! Is it problematic that Turf is tailored for GeoJSON? I'm imagining values outside of [-360, 360] or even cases where Turf is attempting to use a WGS84 ellipsoid.

Reply:
I haven't run into any issues thus far, and it provides a lot of nice functionality so I don't have an issue using it